### DISCUSSION TITLE
Build with Visual Studio 2015

### DISCUSSION THREAD

**Discussion by rlopeznegrete** [shibatch/sleef#430] [Q&A]
Hi all,

Anyone know if it is possible to build the library using visual studio 2015? I am able to configure through cmake, but when building I'm getting some errors related to c starndards. For example 

\src\libm\sleefsimddp.c(2051): error C2059: syntax error: 'bad suffix on number' [C:\Work\CasADI\sleef\build_msvc\src\libm\sleefsse4.vcxproj]

the offending line says
  vopmask o = vlt_vo_vd_vd(d, vcast_vd_d(SLEEF_DBL_MIN));

and the macro SLEEF_DBL_MIN is defined as
#define SLEEF_DBL_MIN 0x1p-1022

The last one being what I think is a standard that Visual Studio 2015 cannot handle. 

Any thoughts on how to correct this besides moving to a newer version of Visual Studio?
Thanks,
  Rodrigo

**Comment by blapie** [comment]
@shibatch Does this relates to the fix you just submitted (#587)? Or is this one specific to older MSVC?

We are looking to add some checks for MSVC, so we might be able to reproduce this sort of issues soon. #588

**Comment by blapie** [comment]
@rlopeznegrete Sorry for the hugely delayed response by the way.

**Reply by rlopeznegrete** [reply]
No worries, I've since moved on and built the whole thing in Linux. However, if there's a fix or requirement to use a newer version of MSVS it would be good to know.

thanks again!

**Comment by shibatch** [comment]
According to the following page, hexadecimal notation for floating-point constants has been supported since Visual Studio 2017.

https://stackoverflow.com/questions/18180116/vc-rejecting-hexadecimal-floating-point-constant

**Reply by blapie** [reply]
Ok, and we don't claim to support earlier than Visual Studio 2019 on https://sleef.org/.
@shibatch Can we claim we support VS 2017? Have you been able to try?

**Reply by shibatch** [reply]
We need to pay for the OS license just to create the environment to build.
MSVC has a different OS structure than others, so it takes a lot of time and effort to support it.
Rather, we should have a policy of not supporting anything other than the latest version of MSVC.
